web/src/LogPane.tsx

```diff
@@ -53,6 +53,52 @@ class LogPane extends Component<LogPaneProps, LogPaneState> {
     }
   }
 
+  componentDidUpdate(prevProps: LogPaneProps) {
+    if (this.props.highlightsEnabled && this.props.modalIsOpen) {
+      document.removeEventListener(
+        "selectionchange",
+        this.handleSelectionChange
+      )
+    }
+
+    if (
+      this.props.highlightsEnabled &&
+      prevProps.modalIsOpen &&
+      !this.props.modalIsOpen
+    ) {
+      document.addEventListener("selectionchange", this.handleSelectionChange, {
+        passive: true,
+      })
+    }
+
+    if (this.props.name !== prevProps.name) {
+      let selection = document.getSelection()
+      this.props.handleClearHighlight()
```

Comment:
the way this bounces control back and forth between the child component and the parent component seems hard to reason about and hard to test.

have you considered having the HUD do this in the history.listen handler? that seems a lot closer to what you want - clearing the selection every time the URL changes.

Reply:
Oh, I was looking for something which would reliably fire when navigating between resources. But didn't realize it was right here. Thanks for pointing it out!